### DISCUSSION TITLE
One version policy and top-level dev dependencies within a pnpm workspace.

### DISCUSSION THREAD

**Discussion by SmashingQuasar** [pnpm/pnpm#6523] [Q&A]
Hey! :wave: 

So I am currently attempting to setup a project using `pnpm workspace` to build a mono-repository but I have been banging my head against the wall for quite some time now.

I'd like all relevant `devDependencies` to be declared at the top-level. This would be `CSpell`, `ESlint`, `ShellCheck` and so on.
I installed them using `pnpm install -w -D` and it did add them at top level. The thing is, when attempting to lint a package using `ESlint`, it fails because it can't resolve `@types/mocha` (linting the unit test here). This dependency is installed but at top level.

I learned that `pnpm overrides` are a thing. However I don't know if that would really solve my one version policy concern. This is because it looks like a very manual approach and it does not seem possible to automate version upgrades through `Dependabot`. If it is fully manual then it's not what I am looking for.

I know how to enforce one version policies using `yarn` through constraints but this is a feature that `pnpm` does not seem to have.

I did find an answer within [StackOverflow](https://stackoverflow.com/questions/65602563/possible-to-use-only-one-package-version-using-pnpm) made by Zoltan Kochan but unfortunately it's pretty scarce and talks about `pnpm overrides` without much substance.

I really have no idea how to handle this. I tried asking on Discord but I never got an answer so I am trying this platform.

Thanks a lot for your help!

**Comment by jaxn** [comment]
You still add the `devDependencies` to each project, but in the root `package.json` you set the overrides to the version you want to use.  Then every project in the monorepo will use the same version.

**Comment by SmashingQuasar** [comment]
@jaxn Thanks for your answer!

I forgot that I had opened this discussion, sorry.
I figured it out by simply adding the `devDependencies` at root level and not within packages. I had a conflict going on with `Nx` so `PNPM` was confused where to look up the dependencies but now it is working.
Regarding overrides, I think they don't meet my needs because this is a manual approach and I needed Dependabot to be able to automatically update the dependencies so it wouldn't work.
